### PR TITLE
Fix loading tiff file lazily

### PR DESCRIPTION
**Pull request by ericpre**
Fix for https://github.com/hyperspy/rosettasciio/issues/316: using the context manager to load tiff files was closing the file. Some tiff were still loading lazily (for example the file used to test lazy loading in the test suite before this PR) because a different code path was being used.

### Progress of the PR
- [x] Fix loading tifffile lazily,
- [x] add support for tiff file in `get_file_handle` 
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add a changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting of the changelog entry (and eventual user guide changes) in the `docs/readthedocs.org:rosettasciio` build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: rsciio/utils/tools.py [hunk 1]
```diff
@@ -503,27 +503,59 @@ def ensure_unicode(stuff, encoding="utf8", encoding2="latin-1"):
 
 
 def get_file_handle(data, warn=True):
-    """Return file handle of a dask array when possible; currently only hdf5 file are
-    supported.
     """
-    arrkey = None
+    Return file handle of a dask array when possible.
```
Comment (CSSFrancis) [inline rsciio/utils/tools.py:507]: How does this work with memmaped files? 

We should make a new issue to maybe make this a bit more routine.

Comment (CSSFrancis) [reply rsciio/utils/tools.py:507]: What if we put the original file into the metadata or added it to the signal dictionary.  Would that be slightly better as we could just track the initial object.

**Review by CSSFrancis — APPROVED** [review]
This is a good change, although what happens in the case of:


```python
s = hs.load("file.tiff", lazy=True)
summed = s.sum()
summed.compute() # This shouldn't close the file correct?
```


```python
s = hs.load("file.tiff", lazy=True)
summed = s.sum()
s.compute()
summed # This should still be lazy with the file not closed, correct?
```

**Comment by ericpre** [comment]
The API to close file is in [`compute(close_file=True)`](https://hyperspy.org/hyperspy-doc/current/reference/base_classes/signal/LazySignal.html#hyperspy._signals.lazy.LazySignal.compute) with the default set to False.

> This is a good change, although what happens in the case of:
> 
> ```python
> s = hs.load("file.tiff", lazy=True)
> summed = s.sum()
> summed.compute() # This shouldn't close the file correct?
> ```
Yes, this will not close the file.

> ```python
> s = hs.load("file.tiff", lazy=True)
> summed = s.sum()
> s.compute()
> summed # This should still be lazy with the file not closed, correct?
> ```
Same as above, this will not close the file.

File: rsciio/utils/tools.py (hunk 1 above)
Comment (ericpre) [reply rsciio/utils/tools.py:507]: Agree, we need an API to do that.

**Pull request merged**